### DISCUSSION TITLE
Merge inertiajs/progress into core

### DISCUSSION THREAD

**Discussion by innocenzi** [inertiajs/inertia#707] [Ideas]
## Idea 

Basically, when PRing https://github.com/inertiajs/progress/pull/18, I thought that it would be nice to have the ability to initialize Inertia like the following: 

```ts
import { createApp, h } from 'vue';
import { createInertiaApp } from '@inertiajs/inertia-vue3';

createInertiaApp({
    progress: { color: '#4B5563' },
    resolve: (name: string) => import(`./pages/${name}`),
    setup: ({ el, app, props, plugin }) => createApp({ render: () => h(app, props) })
        .use(plugin)
        .mixin({ methods: { route } })
        .mount(el)
});
```

Currently, we need to add the progress plugin, to import it and initialize it:

```ts
import { createApp, h } from 'vue';
import { createInertiaApp } from '@inertiajs/inertia-vue3';
import { InertiaProgress } from '@inertiajs/progress';

createInertiaApp({
    resolve: (name: string) => import(`./pages/${name}`),
    setup: ({ el, app, props, plugin }) => createApp({ render: () => h(app, props) })
        .use(plugin)
        .mixin({ methods: { route } })
        .mount(el)
});

InertiaProgress.init({ color: '#4B5563' })
```

Given that 90% of the time, we just use the defaults, I think the progress plugin could just be baked in the core. It's [super lightweight](https://bundlephobia.com/result?p=@inertiajs/progress@0.2.5) anyway. 

Basically, the `createInertiaApp` would have a `progress` property which would be used to initialize the progress plugin. I'm thinking it should be enabled by default, so we could use `progress: false` when it's not wanted, and configure it with `progress: {...options}` when necessary (often).

## Pros

- No longer need to `npm i @inertiajs/progress`
- Less boilerplate code
- Pretty elegant? 

## Cons

- Would need a bit more code in each adapter

**Comment by innocenzi** [comment]
@dacoto97 would love to hear your thoughts about that. :)

**Reply by dacoto** [reply]
I don't use @inertiajs/progress, and neither do many users, so it seems ideal to have the two libraries decoupled and to be able to use the library that each one wants.

**Reply by innocenzi** [reply]
Even if the footprint of the progress component baked into the core is so low, and it's not enabled by default? I would tend to say that the pros outweigh the cons, and people who, like you, don't use it, would not be affected at all.
